lib/charms/opensearch/v0/helper_charm.py

```diff
@@ -115,3 +118,22 @@ def relation_departure_reason(charm: CharmBase, relation_name: str) -> RelDepart
         return RelDepartureReason.SCALE_DOWN
 
     return RelDepartureReason.REL_BROKEN
+
+
+def trigger_peer_rel_changed(
```

Comment:
This is a potential source of noise, why is this needed?

Reply:
It's needed for 2 things: 
1. "roles changes" when manually requested by user (config change) - this is invoked, and triggers a peer rel changed event, where the whole topology is again recomputed and units restart
2. `health.apply(app=True)`, when called from a non leader unit - this unit computes the health but since it cannot write on the app status, it will trigger a peer rel changed event to be caught by the leader who will then compute the status and display it on the app level

Reply:
If we are so coupled, why not running it all under the same config-changed?

Reply:
If these events are so entangled together, why not running it all under the same config-changed?

Reply:
because this coming from when we were doing automated roles-rebalancing, which we're no longer doing sincea couple of weeks. This is something we're planning to address in the next cycle